example/test_problem/Hydro/CCSN/python_script/convert_progenitor_form.py

```diff
@@ -0,0 +1,150 @@
+
+    outer_radius = 0.0
+    outer_vel    = 0.0
+    for line in f:
+        if 'wind:' in line:
+            break
+
+        line_split = line.split()
+        for idx, icol in enumerate(TargetCols):
+            variables[idx] = float(line_split[icol])
+            # convert cell outer variables to cell centered values (radius, velocity)
+            if idx == 0:
+                variables[idx], outer_radius = 0.5 * (outer_radius + variables[idx]), variables[idx]
+            if idx == 3:
+                variables[idx], outer_vel = 0.5 * (outer_vel + variables[idx]), variables[idx]
+
+        GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM, variables])
+
+    # add a safer line for the innermost radius
+    GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM[0], GAMER_DATA_FORM])
+    GAMER_DATA_FORM[0, 0] = 1.0
+
+    return f_info, GAMER_DATA_FORM
+
+
+def convert_MESA(file):
+    # physical constants
+    R_sol = 6.957e10 # solar radius
+    M2CM  = 1e2      # meter to centimeter
+
+    # read the header
+    f_info = ''
+    for i in range(5):
+        f.readline()
+    headers = f.readline().split()
+
+    # find and read targeting variables (radius, density, temperature, radial velocity,
+    #                                    electron fraction, pressure, and angular velocity)
+    nvars = 7
+    TargetCols = np.empty(nvars, dtype=np.int)
+    keys = ["logR", "density", "temperature", "velocity",
+            "ye", "pressure", "omega"]
+    TargetCols = [headers.index(key) for key in keys]
+
+    # GAMER-formatted array
+    GAMER_DATA_FORM = np.empty((0, nvars), dtype=np.float)
+    variables       = np.empty(nvars,      dtype=np.float)
+
+    for line in f:
+        line_split = line.split()
+        for idx, icol in enumerate(TargetCols):
+            variables[idx] = float(line_split[icol])
+            # unit conversion (radius, velocity)
+            if idx == 0:
+                variables[idx] = R_sol * np.power(10, variables[idx])
+            if idx == 3:
+                variables[idx] = M2CM * variables[idx]
+
+        GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM, variables])
+
+    GAMER_DATA_FORM = np.flipud(GAMER_DATA_FORM)
+
+    # add a safer line for the innermost radius
+    GAMER_DATA_FORM = np.vstack([GAMER_DATA_FORM[0], GAMER_DATA_FORM])
+    GAMER_DATA_FORM[0, 0] = 1.0
+
+    return f_info, GAMER_DATA_FORM
+
+
+if __name__ == "__main__":
+    # progenitor model format
+    WHS  = '1'
+    MESA = '2'
+
+    # load the command-line parameters
+    parser = argparse.ArgumentParser(description='Convert the progenitor model to GAMER-supported format\n')
+
+    parser.add_argument('-f', action='store', required=True, type=str, dest='format',
+                        help='original progenitor file format [1: Heager2005, WH07, Sukhbold2015 2: MESA] [%(default)s]')
+    parser.add_argument('-i', action='store', required=True, type=str, dest='input_file',
+                        help='input progenitor file path [%(default)s]')
+
```

Comment:
Another feature is that the progenitors are usually super-giant with r > 1e13 cm, but we only need the inner ~2.e9 cm for the CCSN simulation. We could add another input argument `--rmax` to set the maximum radius included in the GAMER model. The default value could be 1e10.

Reply:
I think we are good to have this feature. I have added another input argument `-rmax` to control the maximum radius of the output file.

Reply:
should it be `--rmax` or `-r` not `-rmax` ?

Reply:
you should check the correct `parser` usage. 
it should be something like  `parser.add_argument("-v", "--verbose", ...`

Reply:
I have changed the `parser arguments`